### PR TITLE
Implement serializer class

### PR DESCRIPTION
**Pull request by byronlin-bq**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: IonHashDotnet/Serializer.cs [hunk 1]
```diff
@@ -1,56 +1,104 @@
...
-        private bool hashContainerAnnotation;
+        private bool hasContainerAnnotation = false;
 
         internal Serializer(IIonHasher hashFunction, int depth)
         {
+            this.HashFunction = hashFunction;
+            this.Depth = depth;
         }
 
+        internal int Depth { get; private set; }
```
Comment (pbcornell) [inline IonHashDotnet/Serializer.cs]: i think this can be private

File: IonHashDotnet/Serializer.cs [hunk 2]
```diff
@@ -85,41 +133,160 @@ private static void Serializers(IonType type, dynamic value, IIonWriter writer)
...
+                {
+                    tq |= 0x0F;
+                }
+            }
+
+            // not a bool, symbol, or null value
+            if (type != IonType.Bool && type != IonType.Symbol && (tq & 0x0F) != 0x0F)
+            {
+                // zero - out the L nibble
+                tq &= 0x0F;
```
Comment (pbcornell) [inline IonHashDotnet/Serializer.cs]: this should be `tq &= 0xF0`

File: IonHashDotnet/Serializer.cs (hunk 1 above)
Comment (simonz-bq) [reply IonHashDotnet/Serializer.cs]: Done.

File: IonHashDotnet/Serializer.cs (hunk 2 above)
Comment (simonz-bq) [reply IonHashDotnet/Serializer.cs]: Great catch, thanks! Fixed.

**Pull request merged**